apps/blog/src/app/form/form.component.ts

```diff
@@ -7,9 +24,66 @@ import { Component, OnInit } from '@angular/core';
 })
 export class FormComponent implements OnInit {
 
-  constructor() { }
+  myform: FormGroup;
+  title: FormControl;
+  author: FormControl;
+  text: FormControl;
+  date: Date;
+  post: Observable<Post>;
+  // posts$: Observable<Post[]>;
+  statusMessage = '';
+  error = false;
+
+  constructor(private http: HttpClient,
+    private formService: FormService,
+    private activatedRoute: ActivatedRoute,
+    private router: Router
+    ) {
+      // this.posts$ = this.formService.repo$.valueChanges();
+     }
 
   ngOnInit() {
+    this.createFormControls();
+    this.createForm();
+    // console.l
+  }
+
+  createFormControls() {
+    this.title = new FormControl('', Validators.required);
```

Comment:
You can create formControls inline above, e.g. on line 28:
title = new FormControl('', Validators.required);

Reply:
Inline? Where?